test/Tx/Spec.hs

```diff
@@ -0,0 +1,216 @@
+module Tx.Spec (tests) where
+
+import Cardano.Api
+import Cardano.Api.Shelley (ProtocolParameters (protocolParamMaxTxExUnits))
+import Data.Aeson (eitherDecodeFileStrict')
+import Data.Bifunctor (first)
+import Data.ByteString (ByteString)
+import Data.ByteString.Base16 qualified as Base16 (encode)
+import Data.Maybe (fromMaybe)
+import Data.Text (Text)
+import Test.Tasty
+import Test.Tasty.HUnit
+import Prelude
+
+import HydraRPS.Tx
+
+import Tx.DummyContract (dummyValidator, dummyValidatorAddress)
+
+tests :: TestTree
+tests =
+  testGroup
+    "TxBody construction"
+    [ testCase "lock to a script output" $
+        (serialiseToCBORHex <$> lockTxBodyBuilder) @?= Right lockTxBodyCBOR
+    , testCase "spend a script output" $ do
+        pparamsResult <- eitherDecodeFileStrict' "devnet/protocol-parameters.json"
+        let txBodyResult = pparamsResult >>= spendTxBodyBuilder
+        (serialiseToCBORHex <$> txBodyResult) @?= Right spendTxBodyCBOR
+    , testCase "sign a transaction" $
+        (serialiseToCBORHex . signTx skey <$> lockTxBodyBuilder) @?= Right lockSignedTxCBOR
+        {-
+          -- a more realistic example with the key in a file:
+          testCaseInfo "sign a transaction" $ do
```

Comment:
Solely because I haven't learnt yet what is the proper way to use data files in tests: where to put them in the project tree, what paths to use to reference them, etc.

Reply:
It would probably be better also to externalize all the tx blobs.